grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/GrabbitExecutionContextDao.groovy

```diff
@@ -0,0 +1,24 @@
+package com.twcable.grabbit.spring.batch.repository
+
+import org.springframework.batch.core.repository.dao.ExecutionContextDao
+import org.springframework.batch.item.ExecutionContext;
+
+
+/**
+ * Modified DAO Interface for persisting and retrieving {@link ExecutionContext}
+ * @see ExecutionContextDao for more details
+ */
+interface GrabbitExecutionContextDao extends ExecutionContextDao {
+
+    /**
+     * Returns job execution context paths by comparing "executionId" property on "executionContext/job/<id>" with
+     * "executionId" property on JobExecutions for the @param jobExecutionResourcePaths
+     */
+    public List<String> getJobExecutionContextPaths(List<String> jobExecutionResourcePaths)
```

Comment:
Do you mean passing String paths around? Do you have any suggestions as to how we can avoid that? I had to pass paths around (which come to think of it I can actually just pass the "nodename" around since the parent path will be constant) since if I pass List<Resource> ... I was having issues removing the resources later (basically.. holding a Resource object in memory and then using resourceResolver to remove that resource object .. was not working)